### DISCUSSION TITLE
Phpfastcache class extending

### DISCUSSION THREAD

**Discussion by thomich** [TruCopilot/phpfastcache#830] [General]
Hello

With the Class example: https://github.com/PHPSocialNetwork/phpfastcache/blob/master/docs/examples/extendedPhpFastCache.php i implemented my own cache class which extends PhpfastcacheAbstractProxy.

Alls fine so far, expect the all know PHP notice: Calling many times CacheManager::getInstance() for already instanced...
My own cache class and the constructor:
![Selection_002](https://user-images.githubusercontent.com/4398840/133972066-3daa077c-559f-434f-af03-e66b997c4547.png)

Then i have a own method to get a cached item:
![Selection_003](https://user-images.githubusercontent.com/4398840/133972196-7044f4c2-4f34-4038-abf6-ab440d2a2e9f.png)

Now in my other classes where i need to work with the cache i use the own cache class:

use app\cache;

And then i load it in the construct of that other class to use it:
$this->cache = new cache();

I know, thats exact the thing that will give above PHP notice warning.. i know...
If i read trough the extendedPhpFastCache example.. i only see the own cache class.. but how to call/load it in other files to re-use it without PHP notice warning?

Hints are very welcome ;)

Thanks a lot
thomi

**Comment by Geolim4** [comment]
Hello,

You need to store (statically or in an other way) your cache instance along your script.

Because actually something is calling the CacheManager multiple times with the same settings.

**Comment by thomich** [comment]
hey @Geolim4 

many thanks.. smart hint and solved ;)

regards
thomi
